jenkins/install_jenkins.sh

```diff
@@ -263,6 +277,9 @@ else
   sudo apt-get install jenkins --yes # sometime the first apt-get install jenkins command fails, so we try it twice
 fi
 
+retry_until_successful sudo test -f /var/lib/jenkins/secrets/initialAdminPassword
+retry_until_successful run_util_script "jenkins/run-cli-command.sh" -c "version"
```

Comment:
The `initialAdminPassword` should be generated after the first run of the Jenkins. I added a check here to ensure it is there before we continue with the followed setup.

However, for Jenkins `2.89.1`, we have seen 2 cases where the file is not generated, and Jenkins seems to be down. The root cause is still unknown.

Reply:
This was a security bug introduced in version 2.81 and was fixed in version 2.89.2.

http://seclists.org/oss-sec/2017/q4/395
https://github.com/jenkinsci/jenkins/commit/ccc374a717